Source/Particles/Collision/BinaryCollision/ElasticCollisionPerez.H

```diff
@@ -107,18 +107,39 @@ void ElasticCollisionPerez (
                amrex::max(n1,n2), T_R(-1.0/3.0) );
     lmdD = amrex::max(lmdD, rmin);
 
+#if (defined WARPX_DIM_RZ)
+    T_R * const AMREX_RESTRICT theta1 = soa_1.m_rdata[PIdx::theta];
+    T_R * const AMREX_RESTRICT theta2 = soa_2.m_rdata[PIdx::theta];
+#endif
+
     // call UpdateMomentumPerezElastic()
     {
       int i1 = I1s; int i2 = I2s;
       for (int k = 0; k < amrex::max(NI1,NI2); ++k)
       {
+
+#if (defined WARPX_DIM_RZ)
+          T_R const theta = theta2[I2[i2]]-theta1[I1[i1]];
```

Comment:
Can you add some comments here explaining what this is doing and why it is needed?

Reply:
Sure.
In our 2D-RZ, when particles are having the same `vr`, without this PR, their velocities would change after collision with each other, because they may have different azimuth angles, but their velocities are not supposed to be changed.
To fix this, we find out the angle between the two particles, and rotate one particle to be in parallel with the other particle, such that their velocities won't change after collision if they have the same `vr`. Then, we need to rotate that particle back.

Reply:
Sorry if I missed discussion of this, but is there an understanding of why the particle's velocity changes if the rotation is not done? It seems that the particle velocities will be at arbitrary angles and the collision does not depend on the particle's theta, so it's not clear why it matters.

Reply:
If particles have the same `vr` but different theta, their `vx` and `vy` are different, so the particles' velocities may point to different directions, and a collision between them would change their velocities. Although their velocities may only change slightly and oscillate around their initial values, as shown in the above plot, but we don't want them to change at all.
@RemiLehe Could you explain more if I missed something?

Reply:
This sounds like a colocation problem - particles with similar r and z can collide even though they have different theta and are in fact far apart from each other. The fix here then is to temporarily map the particles into a common theta. Will this be problematic when using higher order RZ modes since the structure in theta is important? Could there be some theta dependence on the velocity distribution that this fix washes out? A harder question is that could there be a theta dependence in density that will affect collision rates?

Reply:
@dpgrote Yes, I think that you are right for higher mode.
The fix implemented in this PR will be enough for one application that we have in mind in the short term, but you are right that it will cause problem with higher modes. I will add this topic on the agenda for the next developer meeting, and we could discuss it more.

Reply:
@dpgrote I'm not very familiar with higher order RZ modes, but I believe what you said is correct. This PR only works for pure 2D-RZ (no higher order modes). @RemiLehe What do you think? At least we should add some warning or assert in the code to make sure people don't use higher order RZ modes with this PR's fix.

Reply:
Ok. It would be good to add some comments here that this is assuming lowest mode only, i.e. axisymmetry. A warning would be good if the collisions are done with higher order modes turned on. Also, a statement should be added to the documentation for the collisions that it is assuming axisymmetry with RZ (in `usage/parameters.rst`).